Mathlib/Data/Nat/Bitwise.lean

```diff
@@ -318,4 +319,58 @@ theorem lt_lxor'_cases {a b c : ℕ} (h : a < lxor' b c) : lxor' a c < b ∨ lxo
   (or_iff_right fun h' => (h.asymm h').elim).1 <| lxor'_trichotomy h.ne
 #align nat.lt_lxor_cases Nat.lt_lxor'_cases
 
+lemma div2_succ_succ (x) :
+    div2 (x + 2) = (div2 x) + 1 := by
+  simp only [div2_succ, bodd_succ, Bool.cond_not]
+  cases bodd x <;> simp only [cond_false, cond_true]
+
+lemma div2_eq_zero {x : ℕ} :
+    div2 x = 0 → x = 0 ∨ x = 1 := by
+  intro h
+  rcases x with ⟨⟩ | ⟨⟩ | x
+  next => apply Or.inl rfl
+  next => apply Or.inr rfl
+  next =>
+    rw[div2_succ_succ] at h
+    contradiction
+
+lemma bitwise'_bit' {f : Bool → Bool → Bool} (a m b n)
+    (ham : m = 0 → a = true) (hbn : n = 0 → b = true) :
+    bitwise' f (bit a m) (bit b n) = bit (f a b) (bitwise' f m n) := by
+  unfold bitwise'
+  rw [binaryRec_eq', binaryRec_eq']
+  · apply Or.inr hbn
+  · apply Or.inr ham
+
+lemma bitwise_eq_bitwise' (f) : bitwise f = bitwise' f := by
+  funext x y
+  have ⟨k, hk⟩ : ∃ k, k = x+y := by use x+y
+  induction' k using Nat.strongInductionOn with k ih generalizing x y
```

Comment:
Does something like this work, and save the need for the `have`?
```suggestion
  induction' h : x + y using Nat.strongInductionOn with k ih generalizing x y
```

Reply:
It does